File: src/main/java/at/fhhagenberg/sqe/api/ElevatorService.java
package at.fhhagenberg.sqe.api;

import at.fhhagenberg.sqe.entity.Elevator;
import java.rmi.RemoteException;
import java.util.List;

public interface ElevatorService {
    public List<Elevator> getAll() throws RemoteException;
    public Elevator get(int elevatorNumber) throws RemoteException;
    public void updateCommittedDirection(Elevator elevator) throws RemoteException;
    public void updateTargetFloor(Elevator elevator) throws RemoteException;
}

File: src/main/kotlin/at/fhhagenberg/sqe/api/ElevatorService.kt
package at.fhhagenberg.sqe.api

import at.fhhagenberg.sqe.entity.Direction
import at.fhhagenberg.sqe.entity.Elevator
import java.rmi.RemoteException

interface ElevatorService {
    @Throws(RemoteException::class)
    fun getAll(): List<Elevator>

    @Throws(RemoteException::class)
    fun get(elevatorNumber: Int): Elevator?

    @Throws(RemoteException::class)
    fun updateTargetFloor(elevator: Elevator, targetFloor: Int)
}